R/qza_to_tse.R

```diff
@@ -0,0 +1,102 @@
+#' Generates a TreeSummarizedExperiment (TSE) object from .qza artifacts
+#'\code{\link[TreeSummarizedExperiment:SummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'
+#' Construct a TreeSummarizedExperiment object from multiple 
+#' qiime2 artifacts (.qza). 
+#' Embedded metadata for provenance is not maintained in this function and 
+#' instead \code{read_qza()} should be used.
+#' 
+#' 
+#' @param features file path for artifact containing a feature (OTU/SV) table
+#' @param tree file path for artifact containing a tree
+#' @param taxonomy file path for artifact containg taxonomy
+#' @param metadata file path for a qiime2-compliant TSV metadata file
+#' @param tmp a temporary directory that the object will be decompressed to.
+#' @return a TreeSummarizedExperiment object
+#' \code{\link[TreeSummarizedExperiment:SummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' 
+#' @import TreeSummarizedExperiment
+#' @importFrom S4Vectors SimpleList DataFrame make_zero_col_DFrame
+#' @importFrom SummarizedExperiment colData colData<-
+#' 
+#' @examples 
+#' # (Data is from tutorial
+#' https://docs.qiime2.org/2020.2/tutorials/moving-pictures/)
+#' 
+#' tse <-qza_to_tse(
+#' features="path_to_table.qza",
+#' tree="path_to_rooted-tree.qza",
+#' taxonomy="path_to_taxonomy.qza",
+#' metadata = "path_to_sample-metadata.tsv"
+#' )
+#' 
+#' @export
+
+qza_to_tse<- function(features,tree,taxonomy,metadata, tmp) {
+    # Input check
+    if(missing(features)){
+        stop("No features have been specified. Please specify one to create a TSE from qza data.")
+    }
+    
+    # If no temporary extract location is specified, use default
+    if(missing(tmp)){tmp <- tempdir()}
+    
+    if(!missing(features)){
+        # Read the qza features as a matrix
+        features <- as.matrix(read_qza(features, tmp=tmp)$data)
+        # Create a list of assays
+        assays <- S4Vectors::SimpleList(counts =features)
+        
+    }
+    
+    
+    if(!missing(taxonomy)){
+        # Read the taxonomy data
+        rowData <- read_qza(taxonomy, tmp=tmp)$data
+        rowData <- parse_taxonomy(rowData)
+        # Convert to S4 DataFrame
+        rowData <- S4Vectors::DataFrame(data.frame(rowData))
+    }
+    else{
+        # Fills the taxonomy with zero's if no taxonomy table is specified 
```

Comment:
I think strictly taken this "Creates empty table with zero columns" instead of filling with zeroes. 
-> correct the explanation.